go/vendor/github.com/keybase/go-updater/watchdog/watchdog.go

```diff
@@ -101,15 +79,57 @@ func (p Program) terminate(log Log) {
 			break
 		}
 	}
-	if !terminatedPidIsStillLockedByFile {
-		// the program updated its own pidfile
+	return terminatedPidIsStillLockedByFile
+}
+
+func (p Program) deletePidFileIfTerminated(terminatedPids []int, log Log) {
+	if p.PidFile == "" {
+		// there is no pidfile on this program
 		return
 	}
-	if err := os.Remove(p.PidFile); err != nil {
-		log.Infof("error deleting pidfile %s: %s", p.PidFile, err.Error())
+	var err error
+	ticker := time.NewTicker(300 * time.Millisecond)
+	for {
+		select {
+		case <-ticker.C:
+			if !pidFileIsDeletable(terminatedPids, p.PidFile, log) {
+				return
+			}
+			err = os.Remove(p.PidFile)
+			if err != nil {
+				log.Infof("unable to delete a pidfile %s: %s", p.PidFile, err.Error())
+				continue
+			}
+			log.Infof("successfully deleted a pidfile %s", p.PidFile)
+			return
+		case <-time.After(3 * time.Second):
```

Comment:
i think you want to create `time.After` outside of the loop, otherwise it's never be hit because it's created to wait 3 seconds on every iteration, but ticker will always be first.

Reply:
see https://github.com/keybase/client/blob/master/go/bind/keybase.go#L438

Reply:
oops. you are right. nice catch. fixing...